### PR TITLE
Update host preparation section for DRBD

### LINKED ISSUE #145: Improve install instructions for host preparation
Added a separate issue for https://github.com/piraeusdatastore/piraeus-operator/issues/142#issuecomment-755211165

> Looks like the instructions are incomplete for the host set-up. Basically the problem is the missing usermode_helper=disabled parameter. On first install it will most likely work, as the module is only loaded when the injector image runs, and that takes care of setting this option. On restart, I guess the module gets loaded sooner, now without this parameter set.

I'm really looking forward to go through it again :)

### PR DESCRIPTION
**Pull request by WanzenBug**
Expand on the two different ways to install DRBD on the cluster.
* Mark injector image as the preferred install method.
* List available injector images.
* Add instructions on how to disable the usermode_helper on host installs.

Fixes #145

### CODE REVIEW

**Comment by WanzenBug** [comment]
@immanuelfodor I'd be happy to receive your input on this, since you brought it up.

File: README.md [hunk 1]
```diff
@@ -19,13 +19,8 @@ associated Helm chart:
...
 The operator can be deployed with Helm v3 chart in /charts as follows:
 
-* Prepare the hosts for DRBD deployment. There are several options:
-  * Install DRBD directly on the hosts as
-    [documented](https://www.linbit.com/drbd-user-guide/users-guide-9-0/#p-build-install-configure).
-  * Install the appropriate kernel headers package for your distribution and
-    choose the appropriate
-    [kernel module injector](doc/helm-values.adoc#operatorsatellitesetkernelmoduleinjectionimage).
-    Then the operator will compile and load the required modules.
+- Prepare the hosts for DRBD deployment. This depends on your Host OS. You can find more information on the available
```
Comment (immanuelfodor) [inline README.md:22]: Nitpick: Host OS -> host OS

File: doc/host-setup.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+## Build and load DRBD using the Kernel Module Injection Image (Preferred)
+
+Every satellite container created by the operator will use an [InitContainer] to ensure DRBD is available. This init
+step can also compile and load DRBD directly. To build DRBD, the container needs access to the kernel sources. It will
+look for the sources on the host under `/usr/src` and `/lib/modules/$(uname -r)`. Most distributions provide a convenient package
+for the kernel source:
+
+* Ubuntu: `apt-get install linux-headers-$(uname -r)`
+* CentOS: `yum install kernel-devel-$(uname -r)`
```
Comment (immanuelfodor) [inline doc/host-setup.md:17]: For me it was called `kernel-devel` on CentOS. I also installed `kernel-headers`, I don't know if it's needed but it seemed obvious to have besides the source files. 

I also installed `cryptsetup` to get the greenLUKS checkmarks in Linstor output but it didn't work, it's always unavailable.

File: doc/host-setup.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+Install these packages and use the injector image matching your distribution:
+
+* CentOS 7: `quay.io/piraeusdatastore/drbd9-centos7`
+* CentOS 8: `quay.io/piraeusdatastore/drbd9-centos8` (**Not** guaranteed to work on CentOS 8 Stream)
```
Comment (immanuelfodor) [inline doc/host-setup.md:22]: A user reported that it works with the following Kernel, more info at: https://github.com/piraeusdatastore/piraeus-operator/issues/137

```
$ uname -r                                                         
4.18.0-240.1.1.el8_3.x86_64
```

Comment (immanuelfodor) [reply doc/host-setup.md:22]: Or: 

See https://github.com/piraeusdatastore/piraeus-operator/issues/137 for more info.

File: doc/host-setup.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
+# Preparing the host for DRBD
+
+Piraeus depends on DRBD for data replication across multiple nodes. [DRBD] is a kernel module that must be loaded on
+every node that should access the storage. Depending on your exact host OS and set up, you can choose from one of the
```
Comment (immanuelfodor) [inline doc/host-setup.md:4]: Maybe more clear: you can choose **one** from the following options

File: doc/host-setup.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+changes in DRBDs state. When using it with containers, it could confuse programs that expect to know of all configured
+DRBD resources, such as the default `drbdadm`. This command then reports an error, which is interpreted by DRBD to abort
+the current transition. In the end, you are left with a lot of `standalone` resources that can't be used.
+
+To prevent this issue, you have to set the DRBD module parameter `usermode_helper` to `disabled`. To check the current
+value, run:
+
+```
+# cat /sys/module/drbd/parameters/usermode_helper
+/sbin/drbdadm
```
Comment (immanuelfodor) [inline doc/host-setup.md:49]: Are you sure the output is the path of the binary, and not enabled? The other example below says disabled. (Mine is disabled, too.)

File: doc/host-setup.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+* CentOS 7: `quay.io/piraeusdatastore/drbd9-centos7`
+* CentOS 8: `quay.io/piraeusdatastore/drbd9-centos8` (**Not** guaranteed to work on CentOS 8 Stream)
+* Ubuntu 20.04: `quay.io/piraeusdatastore/drbd9-focal`
+* Ubuntu 18.04: `quay.io/piraeusdatastore/drbd9-bionic`
+
+You can set the image by passing `--set operator.satelliteSet.kernelModuleInjectionImage=<image>` to Helm.
+
+[InitContainer]: https://kubernetes.io/docs/concepts/workloads/pods/init-containers/
+
+## Install DRBD on the Host
```
Comment (immanuelfodor) [inline doc/host-setup.md:30]: Nitpick: Host -> host

**Review by immanuelfodor — CHANGES_REQUESTED** [review]
This is such a huge improvement from the previous one, great job. I've added small remarks that might need a change, please check them.

File: doc/host-setup.md (hunk 5 above)
Comment (WanzenBug) [reply doc/host-setup.md:50]: Yeah, I'm sure :wink: 

Tested it in a fresh VM. `disabled` is a special string for DRBD in this context

File: doc/host-setup.md (hunk 2 above)
Comment (WanzenBug) [reply doc/host-setup.md:17]: `kernel-devel` always tracks the latest kernel version, which may not be the version currently running, so I added the `$(uname -r)` shortcut.

Haven't checked on LUKS support in a while, I guess that's a separate issue.

File: doc/host-setup.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+they are often out of date and can't be used for Piraeus. Instead, you should follow the [link to DRBDs user guide] to
+ensure you install a recent enough version. After that, please read the instructions below to make DRBD ready for
+containerized workloads.
+
+[link to DRBDs user guide]: https://www.linbit.com/drbd-user-guide/users-guide-9-0/#p-build-install-configure
+
+You have to disable DRBD's user mode helper. This feature of the DRBD module enables running user configured commands on
+changes in DRBDs state. When using it with containers, it could confuse programs that expect to know of all configured
+DRBD resources, such as the default `drbdadm`. This command then reports an error, which is interpreted by DRBD to abort
+the current transition. In the end, you are left with a lot of `standalone` resources that can't be used.
```
Comment (JoelColledge) [inline doc/host-setup.md:43]: Nitpicking: `StandAlone`

**Review by JoelColledge — APPROVED** [review]

**Pull request merged**